cdap-data-fabric/src/main/java/co/cask/cdap/data2/util/hbase/HBaseTableUtil.java

```diff
@@ -385,6 +387,70 @@ public OutputStream getOutput() throws IOException {
     return info;
   }
 
+  protected String toHBaseNamespace(Id.Namespace namespace) {
+    return "cdap_" + namespace.getId();
+  }
+
+  /**
+   * Returns if the current version of HBase supports namespaces
+   *
+   * @return true if namespaces are supported, false otherwise
+   */
+  public abstract boolean namespacesSupported();
+
+  /**
+   * Creates a new {@link HTable} which may contain an HBase namespace depending on the HBase version
+   *
+   * @param conf the hadoop configuration
+   * @param tableId the {@link TableId} to create an {@link HTable} for
+   * @return an {@link HTable} for the tableName in the namespace
+   */
+  public abstract HTable getHTable(Configuration conf, TableId tableId) throws IOException;
+
+  /**
+   * Creates a new {@link HTableDescriptor} which may contain an HBase namespace depending on the HBase version
+   *
+   * @param tableId the {@link TableId} to create an {@link HTableDescriptor} for
+   * @return an {@link HTableDescriptor} for the tableName in the namespace
+   */
+  public abstract HTableDescriptor getHTableDescriptor(TableId tableId);
+
+  /**
+   * Checks if an HBase namespace already exists
+   *
+   * @param admin the {@link HBaseAdmin} to use to communicate with HBase
+   * @param namespace the {@link Id.Namespace} to check for existence
+   */
+  public abstract boolean hasNamespace(HBaseAdmin admin, Id.Namespace namespace);
+
+  /**
+   * Creates an HBase namespace, if it does not already exist
+   * This method uses {@link Id.Namespace} here to cover for a future case when CDAP namespaces may push down namespace
+   * properties to HBase
+   *
+   * @param admin the {@link HBaseAdmin} to use to communicate with HBase
+   * @param namespace the {@link Id.Namespace} to create
+   * @throws IOException if the namespace already exists in HBase
+   */
+  public abstract void createNamespaceIfNotExists(HBaseAdmin admin, Id.Namespace namespace) throws IOException;
+
+  /**
+   * Creates an HBase namespace, if it exists
+   *
+   * @param admin the {@link HBaseAdmin} to use to communicate with HBase
+   * @param namespace the {@link Id.Namespace} to delete
+   * @throws IOException if the namespace does not exist in HBase
+   */
+  public abstract void deleteNamespace(HBaseAdmin admin, Id.Namespace namespace) throws IOException;
+
+  /**
+   * Returns the fully qualified table name containing namespace
+   *
+   * @param tableId the identifier for the table containing namespace and table name
+   * @return the fully qualified table name containing namespace
+   */
+  public abstract String getTableNameWithNamespace(TableId tableId);
```

Comment:
@ghelmling We need this method. We have direct HBaseAdmin usages in the main code in data-fabric (that use table name as a String) as well. I'm going to keep this for now, and see if I get the time to clean up those usages during these two weeks. We will probably need to wrap methods like disableTable, deleteTable, etc in the compat modules eventually.